scripts/verify_adv_IM.py

```diff
@@ -32,79 +66,163 @@ def parse_args():
         help="list of component that ran",
     )
 
+    parser.add_argument(
+        "--station_file",
+        default=None,
+        type=str,
+        help="if set, script will only check for folder-names that match the station file",
+    )
+
+    # saves output
+    parser.add_argument(
+        "--simple_check",
+        default=None,
+        action="store_true",
+        help="a quick comparison of station list, will not generate status.csv",
+    )
+
     args = parser.parse_args()
 
     return args
 
 
-def check_log(list_folders, model, csv_stations, components):
+def check_log(list_folders, model, components, df_model, break_on_fail=False):
     """
     check of "Failed" msg in logs
     no "Failed" in all Analysis*.txt == crashed
     "Failed" in all Analysis*.txt == failed to converge (normal)
     """
-    failed_stations = []
+    #    print(f"{df_model}")
     for station_dir in list_folders:
         # check if folder is a station run folder
         station_model_dir = os.path.join(station_dir, model)
         station_name = os.path.basename(station_dir)
-        if not os.path.isdir(station_model_dir) or (station_name in csv_stations):
-            # does not match naming for a station run, possibly created by something else, skipping
-            continue
         for comp in components:
             component_outdir = os.path.join(station_model_dir, comp)
-            if not check_status(component_outdir, check_fail=True):
-                failed_stations.append(station_name)
-                # logs shows not all analysis failed to converge, but station failed to create csv
-                print(f"{station_dir} failed to run on {model}")
-                # break the loop as soon as one component fail the verification
-                break
-    return failed_stations
-
+            time_list = read_timelog(component_outdir)
+            if check_status(component_outdir):
+                # successed
+                station_component_status = run_status.finished.value
+            # check for 'Failed' keyword
+            elif check_status(component_outdir, check_fail=True):
+                # all logs showed "Failed", analysis was unable to converge
+                station_component_status = run_status.not_converged.value
+            elif time_list.count(None) == 0:
+                # if start and end are there, but no data = crashed
+                station_component_status = run_status.crashed.value
+            elif time_list.count(None) == 1:
+                # only starttime exist = wct timed out
```

Comment:
This makes the assumption that only the first value was found. The check should be made more explicit to verify this. e.g. time_list[0] is not None and time_list[1] is None.
With an error if the other case occurs